web/src/copy_and_paste.ts

```diff
@@ -154,7 +154,43 @@ function remove_div(_div: JQuery, ranges: Range[]): void {
     }, 0);
 }
 
-export function copy_handler(): void {
+async function copy_selection_to_clipboard(selection: Selection): Promise<void> {
+    const range = selection.getRangeAt(0);
+    const div = document.createElement("div");
+    div.append(range.cloneContents());
+    const html_content = div.innerHTML.trim();
+    const plain_text = selection.toString().trim();
+
+    // Reference: https://stackoverflow.com/a/77305170/21940401
+    if (typeof ClipboardItem !== "undefined") {
+        // Shiny new Clipboard API, not fully supported in Firefox.
+        // https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API#browser_compatibility
+        const html = new Blob([html_content], {type: "text/html"});
+        const text = new Blob([plain_text], {type: "text/plain"});
+        const data = new ClipboardItem({"text/html": html, "text/plain": text});
+        await navigator.clipboard.write([data]);
+    } else {
+        // Fallback using the deprecated `document.execCommand`.
+        // https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand#browser_compatibility
+        const cb = (e: ClipboardEvent): void => {
+            e.clipboardData?.setData("text/html", html_content);
+            e.clipboardData?.setData("text/plain", plain_text);
+            e.preventDefault();
+        };
+        document.addEventListener("copy", cb);
+        // eslint-disable-next-line @typescript-eslint/no-deprecated
+        document.execCommand("copy");
+        document.removeEventListener("copy", cb);
+    }
+}
+
+function get_parent(node: Node | null): Element | null {
+    if (node === null || node instanceof Element) {
+        return node;
+    }
+    return node.parentElement;
+}
```

Comment:
This seems like a very strange function, maybe working around a typing issue? It seems very weird you'd potentially return `node` itself in a function called `get_parent`.

Reply:
`get_closest_element` sounds like a function to get the closest element, not to get the thing to call `closest` on.